compact_str/src/features/sqlx.rs

```diff
@@ -41,7 +39,7 @@ where
     DB: Database,
     for<'x> &'x str: Decode<'x, DB> + Type<DB>,
 {
-    fn decode(value: <DB as HasValueRef<'r>>::ValueRef) -> Result<Self, BoxDynError> {
+    fn decode(value: <DB as Database>::ValueRef<'r>) -> Result<Self, BoxDynError> {
         let value = value.to_owned();
```

Comment:
Do we need `to_owned()` here?

Reply:
Good call! I tried removing the `to_owned()` but that didn't work because all we have here is a [`ValueRef`](https://docs.rs/sqlx/latest/sqlx/trait.ValueRef.html). Instead I refactored this to use the `Decode` impl for `&str` which presumably would do the most efficient thing